.pre-commit-config.yaml

```diff
@@ -53,13 +53,27 @@ repos:
     rev: v9.7.0
     hooks:
       - id: commitlint
-        additional_dependencies: ['@commitlint/config-conventional']
+        additional_dependencies: ["@commitlint/config-conventional"]
         stages: [commit-msg]
 
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: 'v1.6.1'
+    rev: "v1.6.1"
     hooks:
       - id: mypy
+        args: [
+            # --ignore-missing-imports is required to run mypy on virtual
+            # environments without installed dependencies
+            "--ignore-missing-imports",
```

Comment:
pre-commit does not install the environment so it can't chase the dependency types (e.g. hail). This setting is different in the TOML that feeds the vscode warnings/errors. VScode is therefore more strict than the pre-commit